modules/reactor/src/meshgenerators/AzimuthalBlockSplitGenerator.C

```diff
@@ -102,8 +102,9 @@ AzimuthalBlockSplitGenerator::generate()
   MeshTools::Modification::rotate(mesh, 90.0, 0.0, 0.0);
   _azimuthal_angle_meta = azimuthalAnglesCollector(mesh, -180.0, 180.0, ANGLE_DEGREE);
   // So that this class works for both derived classes of PolygonMeshGeneratorBase
-  _pattern_pitch_meta = std::max(getMeshProperty<Real>("pitch_meta", _input_name),
-                                 getMeshProperty<Real>("pattern_pitch_meta", _input_name));
+  auto pattern_pitch_meta = std::max(getMeshProperty<Real>("pitch_meta", _input_name),
+                                     getMeshProperty<Real>("pattern_pitch_meta", _input_name));
+  setMeshProperty("pattern_pitch_meta", pattern_pitch_meta);
```

Comment:
It generally looks good to me. Thank you @shikhar413 

Just a comment. It does not occur in this commit. But it is possible that `data_value` of `setMeshProperty()` may have a different type from the originally declared mesh metadata type. So it may be safer to specify the type when calling `setMeshProperty()`

Reply:
That's a good idea. An alternative would be to error if the name is already used with a different data type

Reply:
I did confirm that the user can call `setMeshProperty` by defining `data_value` as an arbitrary type that is different from the originally intended data type and it compiled without error. I do like the idea of adding an error within the function itself as a way to safeguard against a user defining the incorrect datatype from outside. 

@GiudGiud is there a way to modify setMeshProperty to get the original datatype of the restartable data? Right now a `static_cast` is used within setMeshProperty which basically does the datatype conversion at runtime, not at compile time. Changing to `dynamic_cast` didn't fix the issue either, since I was able to define a meta data that was supposed to be Real to std::string

Reply:
There's a prettyCppType routine you can look at how it works to see if we have something, in that same file maybe.

Otherwise one classic way is to check whether a dynamic cast of the pointer passes or fails.

Reply:
So the RestartableData class has a function type() that returns the type of the metadata using prettyCppType, I could have a condition like this within setMeshProperty:

```if (_app.getRestartableMetaData(full_name, MooseApp::MESH_META_DATA, 0).type() != MooseUtils::prettyCppType(&data_value)))
  // Throw error
```

Is that the preferred route? This basically enforces that the same datatype used to define the metadata must be used to redefine it, and conversions from long to int etc. probably won't be allowed since we are comparing to the exact name of the datatype

Reply:
I think it s fair. Too bad for long-int conversions. It s risky anyway so I d rather we don’t support them

Reply:
Check has been added